misc/diagrams.md

```diff
@@ -0,0 +1,29 @@
+# Diagramming Guidelines
+
+We use [Mermaid](https://mermaid.js.org/intro/) as our primary diagramming tool. Mermaid is “a JavaScript based diagramming and charting tool that renders Markdown-inspired text definitions to create and modify diagrams dynamically”. Mermaid enables us to:
+
+* take a “diagram-as-code” approach for creating diagrams;
+
+* include diagrams as part of code review;
+
+* easily modify diagrams to reflect updated functionality;
+
+* maintain change history of our diagrams (i.e., changes tracked via git commits);
+
+* collaborate with one another more effectively when creating diagrams;
+
+* establish consistency in our documentation by using a shared language.
+
+This guideline provides some tips and questions to consider when creating diagrams with accessibility and interpretability in mind.
+
+## Who is the intended audience
+
+Before starting your diagram, it’s important to think about your intended audience. Who are you creating the diagram for and who is it meant to help? Your audience will inform the level of detail your diagram requires. For instance, a low-level, “nuts-and-bolts” diagram may be more useful for InfraEng, while higher-level diagrams may be better suited for other stakeholders.
```

Comment:
Non-blocking: consider changing `InfraEng` to `an engineering team` (or similar) now that this is a shared document rather than an individual team based document.

Reply:
The text has been updated.